dnschain/server.py

```
@@ -26,7 +27,27 @@ def lookup(self, name):
 
         @param name: The name to lookup, e.g. 'id/dionyziz'
         """
-
-        # namecoin_string = read_from_dnschain(name)
-        data = json.loads(namecoin_string)
+        #uname = name.encode("utf-8")#Who knows what bytes will be submitted?! Look up dnschain spec.
+        url_safe_name = urllib.quote(name, safe="") #Meant to be a url path _component_.
+        path = "/id/%s" % (url_safe_name,)
+        self.connection.request("GET", path)
+        namecoin_response = self.connection.getresponse()
+        namecoin_string = namecoin_response.read()
+        try:
+            data = json.loads(namecoin_string)
+        except ValueError, e:
+            if namecoin_string.startswith("Not Found: "):
+                print "The name: %s was not found in the database, returning None." % (url_safe_name)
+                return None
+            else:
+                print "Instead of JSON we got this:\n%s\n" % (e,)
+                import sys
+                sys.exit(99) #Fail hard.
```

Hmm, the library should never quit the app. That is not in its "domain" to do. Let an exception be thrown instead.